src/guest_memory.rs

```diff
@@ -115,9 +123,23 @@ impl Display for Error {
 /// On ARM64, a 32-bit hypervisor may be used to support a 64-bit guest. For simplicity,
 /// `u64` is used to store the the raw value no matter if the guest a 32-bit or 64-bit virtual
 /// machine.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd)]
 pub struct GuestAddress(pub u64);
+
+impl From<u64> for GuestAddress {
+    fn from(value: u64) -> Self {
+        GuestAddress(value)
+    }
+}
+
+impl Into<u64> for GuestAddress {
```

Comment:
And Into should probably never be implemented, instead you should implement `From<GuestAddress> for u64`.